spring/src/main/java/io/kubernetes/client/spring/extended/manifests/KubernetesFromConfigMapProcessor.java

```diff
@@ -0,0 +1,163 @@
+
+  private static final Logger log = LoggerFactory.getLogger(KubernetesFromConfigMapProcessor.class);
+
+  private ApplicationContext applicationContext;
+
+  private final ScheduledExecutorService configMapKeyRefresher =
+      Executors.newSingleThreadScheduledExecutor();
+
+  @Autowired private KubernetesManifestsProperties manifestsProperties;
+
+  public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+
+    for (Field field : bean.getClass().getDeclaredFields()) {
+      ReflectionUtils.makeAccessible(field);
+      try {
+        if (field.get(bean) != null) {
+          continue; // field already set, skip processing
+        }
+      } catch (IllegalAccessException e) {
+        log.warn("Failed inject resource for @FromConfigMap annotated field {}", field, e);
+        continue;
+      }
+
+      FromConfigMap fromConfigMapAnnotation = field.getAnnotation(FromConfigMap.class);
+      if (fromConfigMapAnnotation == null) {
+        continue; // skip if the field doesn't have the annotation
+      }
+
+      if (!Map.class.isAssignableFrom(field.getType())) {
+        log.warn(
+            "Failed inject resource for @FromConfigMap annotated field {}, the declaring type should be Map<String, String>",
+            field);
+        continue;
+      }
+
+      ConfigMapGetter configMapGetter =
+          getOrCreateConfigMapGetter(fromConfigMapAnnotation, applicationContext);
+
+      LoadingCache<String, String> configMapDataCache =
+          Caffeine.newBuilder()
+              .expireAfterWrite(manifestsProperties.getRefreshInterval())
+              .build(
+                  new ConfigMapGetterCacheLoader(
+                      () -> {
+                        return configMapGetter.get(
+                            fromConfigMapAnnotation.namespace(), fromConfigMapAnnotation.name());
+                      }));
+      fullyRefreshCache(configMapGetter, fromConfigMapAnnotation, configMapDataCache);
+      configMapKeyRefresher.scheduleAtFixedRate(
+          () -> {
+            fullyRefreshCache(configMapGetter, fromConfigMapAnnotation, configMapDataCache);
+          },
+          manifestsProperties.getRefreshInterval().getSeconds(),
+          manifestsProperties.getRefreshInterval().getSeconds(),
+          TimeUnit.SECONDS);
+      ReflectionUtils.setField(field, bean, configMapDataCache.asMap());
+    }
+
+    return bean;
+  }
+
+  private static void fullyRefreshCache(
+      ConfigMapGetter configMapGetter,
+      FromConfigMap fromConfigMapAnnotation,
+      LoadingCache<String, String> configMapDataCache) {
+    V1ConfigMap configMap =
+        configMapGetter.get(fromConfigMapAnnotation.namespace(), fromConfigMapAnnotation.name());
+    if (configMap == null || configMap.getData() == null) {
+      return;
+    }
+    configMap.getData().keySet().stream().forEach(key -> configMapDataCache.refresh(key));
```

Comment:
Do we want an update to be atomic? e.g. it is currently possible that a user could read half the values from ConfigMap v1 and half the values from ConfigMap v2.

This would probably be pretty unexpected from a programmers perspective.

Reply:
i added a todo comment on the line. am not sure if atomic cache entry update is supported in the cache library for now, we will need that otherwise it wont work